staging/src/k8s.io/apimachinery/pkg/apis/meta/v1alpha1/types.go

```diff
@@ -60,13 +68,54 @@ type TableRow struct {
 	// cells will be as wide as headers and may contain strings, numbers, booleans, simple maps, or lists, or
 	// null. See the type field of the column definition for a more detailed description.
 	Cells []interface{} `json:"cells"`
+	// conditions describe additional status of a row that are relevant for a human user.
+	// +optional
+	Conditions []TableRowCondition `json:"conditions,omitempty"`
 	// This field contains the requested additional information about each object based on the includeObject
 	// policy when requesting the Table. If "None", this field is empty, if "Object" this will be the
 	// default serialization of the object for the current API version, and if "Metadata" (the default) will
 	// contain the object metadata. Check the returned kind and apiVersion of the object before parsing.
-	Object runtime.RawExtension `json:"object"`
+	// +optional
+	Object runtime.RawExtension `json:"object,omitempty"`
+}
+
+// TableRowCondition allows a row to be marked with additional information.
+// +protobuf=false
+type TableRowCondition struct {
+	// Type of row condition.
+	Type RowConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status ConditionStatus `json:"status"`
+	// (brief) machine readable reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty"`
 }
 
+type RowConditionType string
+
+// These are valid conditions of a row. This list is not exhaustive and new conditions may be
+// inculded by other resources.
+const (
+	// RowCompleted means the underlying resource has reached completion and may be given less
+	// visual priority than other resources.
+	RowCompleted RowConditionType = "Completed"
```

Comment:
I was actually trying to avoid universal, at least in the "if we add a new condition, it's easy enough for a client to decide to add special logic for it".  Having multiple conditions for different resource types could be acceptable.

I was thinking completed because we have other things that can complete, and it's actually not a hard boundary (filtering a pod that just completed is kind of dumb).  So I had been thinking about transition time.

Also, active is a positive condition we would apply to most things, which means we clutter every row with it.  Advantage of calling out terminal is that it is usually the exceptional case.